src/coreclr/jit/ifconversion.cpp

```diff
@@ -140,7 +140,16 @@ bool OptIfConversionDsc::IfConvertCheckThenFlow()
             if (thenBlock->KindIs(BBJ_RETURN))
             {
                 assert(m_finalBlock == nullptr);
-                m_mainOper = GT_RETURN;
+#ifdef SWIFT_SUPPORT
+                if (m_comp->lvaSwiftErrorArg != BAD_VAR_NUM)
+                {
+                    m_mainOper = GT_SWIFT_ERROR_RET;
+                }
+                else
+#endif // SWIFT_SUPPORT
+                {
+                    m_mainOper = GT_RETURN;
+                }
```

Comment:
Are these changes necessary?
It looks like `IfConvertCheckStmts` will fail naturally for `GT_SWIFT_ERROR_RET`, so perhaps no changes in if-conversion are necessary?

Reply:
You're right, we don't check `m_mainOper` until after we call `IfConvertCheckStmts`, and we bail if the terminator node is a `GT_SWIFT_ERROR_RET`, so it should be safe to undo these changes.